### DISCUSSION TITLE
How to improve the network performance of mtg?

### DISCUSSION THREAD

**Discussion by CCCAUCHY** [9seconds/mtg#251] [Q&A] [answered]
How to improve the network performance of mtg? especially on iOS

My server's performance is sufficient

**Comment by 9seconds** [comment, marked as answer]
This is not about server performance per se. I suppose, your complain is about videos or other media. If you use Telegram without proxies, it can access CDNs with these assets. But in case of proxy, these RPC calls are not used and Telegram client fallbacks to an old method of fetching this data that is not that efficient. I do not know what to do here.

You can try other alternatives:

- https://github.com/alexbers/mtprotoproxy
- https://github.com/seriyps/mtproto_proxy

It could be, they are more efficient in your case 🙏

**Reply by CCCAUCHY** [reply]
It seems that the Pyhton version and the Erlang version can access CDNs.

Do you recommend [the C version](https://github.com/TelegramMessenger/MTProxy)?

**Reply by 9seconds** [reply]
It seems that middle proxies work better here than direct connectivity :/ You can try C one. C is the official one

**Reply by CCCAUCHY** [reply]
Is the C version middle proxy or direct connectivity?

**Reply by 9seconds** [reply]
Middle one. If you have adtag, you are probably going to use middle proxies. At least, all implementations except of mtg work like this.
